public/app/features/templating/template_srv.ts

```diff
@@ -419,15 +431,15 @@ export class TemplateSrv {
     }
 
     if (getConfig().featureToggles.newVariables && !this.index[name]) {
-      return getVariableWithName(name);
+      return this.dependencies.getVariableWithName(name);
     }
 
     return this.index[name];
   };
 
   private getAdHocVariables = (): any[] => {
     if (getConfig().featureToggles.newVariables) {
-      return getFilteredVariables(isAdHoc);
+      return this.dependencies.getFilteredVariables(isAdHoc);
```

Comment:
Just a thought, this will be in unsorted order.

Reply:
hmm - good point. Need to have a look at where it is used.